tests/foreman/cli/test_computeresource_azurerm.py

```diff
@@ -528,3 +545,122 @@ def test_positive_azurerm_host_provisioned(
         # Azure cloud
         assert self.hostname.lower() == azureclient_host.name
         assert self.vm_size == azureclient_host.type
+
+
+@run_in_one_thread
+class TestAzureRm_BYOS_FinishTemplate_Provisioning:
+    """ AzureRM Host Provisioning Test with BYOS Image
+
+    """
+
+    @pytest.fixture(scope='class', autouse=True)
+    def class_setup(
+        self, request, module_domain, module_azurerm_cr, module_azurerm_byos_finishimg
+    ):
+        """
+        Sets Constants for all the Tests, fixtures which will be later used for assertions
+        """
+        request.cls.region = settings.azurerm.azure_region
+        request.cls.rhel7_ft_img = AZURERM_RHEL7_FT_BYOS_IMG_URN
+        request.cls.rg_default = AZURERM_RG_DEFAULT
+        request.cls.premium_os_disk = AZURERM_PREMIUM_OS_Disk
+        request.cls.platform = AZURERM_PLATFORM_DEFAULT
+        request.cls.vm_size = AZURERM_VM_SIZE_DEFAULT
+        request.cls.hostname = gen_string('alpha')
+        request.cls.fullhostname = f'{self.hostname}.{module_domain.name}'.lower()
+
+        request.cls.compute_attrs = (
+            f'resource_group={self.rg_default},vm_size={self.vm_size}, '
+            f'username={module_azurerm_byos_finishimg.username}, '
+            f'ssh_key_data={settings.azurerm.ssh_pub_key}, platform={self.platform},'
+            f'script_command={"touch /var/tmp/test.txt"}, script_uris={AZURERM_FILE_URI},'
+            f'premium_os_disk={self.premium_os_disk}'
+        )
+        nw_id = module_azurerm_cr.available_networks()['results'][-1]['id']
+        request.cls.interfaces_attributes = (
+            f'compute_network={nw_id},compute_public_ip=Static, compute_private_ip=false'
+        )
+
+    @pytest.fixture(scope='class')
+    def class_host_ft(
```

Comment:
```suggestion
    def class_byos_ft_host(
```
Putting more weightage on type of host i.e `byos_ft_host`.